pkg/node/utils.go

```diff
@@ -69,26 +70,29 @@ func FindDrives(ctx context.Context, nodeID string, procfs string) ([]*direct_cs
 			return filepath.SkipDir
 		}
 
-		link, err := os.Readlink(path)
+		realpath, err := filepath.EvalSymlinks(path)
```

Comment:
EvalSymlinks is the correct usage @haslersn thanks for the fix

Reply:
It doesnt add any value because absolute paths are not necessary here. How is this the correct fix?

Reply:
It is necessary to use the realpath in `visited` because otherwise there might be an endless loop in case of cyclic symlinks (or rather it fails at some depth where there are too many symlinks).

Reply:
The problem is cyclical links are possible with symlinks - ReadLink is not designed for that https://golang.org/pkg/path/filepath/#EvalSymlinks is the correct from an expectation point of view.

Reply:
I'm well aware of cyclical links. In this case, since we are not recursing unboundedly, it is impossible. Now, if it was possible that there were cyclical links. EvalSymlinks would still not help. Because the in visited[path], the path, which would be a result of EvalSymlinks is still a relative path. 

I'm not sure what prompted this PR, since it does not fix a problem. If the kubelet is thrashing because of recursive calls from a completely different process, which would run out of memory if this infinite recursion happened, and not affect the kubelet. There is no way direct-csi would affect kubelet directly.

Just to be thorough. We have been testing this on mutliple environments as a part of our testing to ensure that it works with multiple versions of kubernetes. This issue has never occurred.

Reply:
From what I understood from this PR - this was a defensive change @wlan0 - not sure if there is a valid error that this PR fixes perhaps @haslern should fill us in. 

It is important to note that ReadLink should never be used in a naked manner, generally it is known to cause cyclical loops - EvalSymlinks simply traverses everything to give you a path that doesn't result in ELOOP - i.e ELOOP is returned if indeed that is going to happen eventually - ReadLink, on the other hand, it won't return such an error until you reach a weird state recursively going deeper.

So from what I saw it is a defensive code to avoid naked ReadLink usage.